src/Approximations/decompositions.jl

```diff
@@ -255,6 +256,40 @@ function decompose(S::LazySet, block_options; block_size::Int=1)
     return decompose(S, partition, block_options)
 end
 
+"""
+    project(S::LazySet{N},
+            block::AbstractVector{Int},
+            ::Nothing,
+            [n]::Int=dim(S)
+           ) where {N<:Real}
+
+Project a high-dimensional set to a given block by using a concrete linear map.
+
+### Input
+
+- `S`       -- set
+- `block`   -- block structure - a vector with the dimensions of interest
+- `nothing` -- used for dispatch
+- `n`       -- (optional, default: `dim(S)`) ambient dimension of the set `S`
+
+### Output
+
+A set representing the projection of the set `S` to block `block`.
+
+### Algorithm
+
+We apply the function `linear_map`.
+"""
+@inline function project(S::LazySet{N},
+                         block::AbstractVector{Int},
+                         ::Nothing,
```

Comment:
The call `decompose(b, partition, nothing)` with a `nothing` is strange.. can we make this the default, so we don't have to enter the dummy argument?

Reply:
The issue #1726 that this PR closes wants to offer a concrete `decompose`. The `project` methods are called from `decompose` by just passing its argument, and it is not possible to have no argument in `decompose`.

https://github.com/JuliaReach/LazySets.jl/blob/da8213f0ba11c9b1513e5f8100378db0ea3af444/src/Approximations/decompositions.jl#L221-L229

We wanted to add the default in #1331. But I agree that the change is simple and I can do it here already.

Reply:
Ok. We can review the interface of decompose later. Also, i remember that it was type unstsble for a reachability use case.

About the default, here or later as you prefer.

Reply:
It was a good idea, I did not see that.
```julia
julia> B = rand(BallInf)
BallInf{Float64}(t[-0.41391500435046535, 0.9008017376015398], 0.25252464855799645)                                             
                                                                                                                                     
julia> project(B, [1])
Zonotope{Float64}([-0.41391500435046535], [0.25252464855799645])                                                                     
```